packages/core/src/test-data/Loader.ts

```diff
@@ -67,56 +145,96 @@ export class DataLoader<T> extends Loader<T> {
 }
 
 export class JSONLoader<T> extends Loader<T> {
+	constructor(public filePath: string) {
+		super(filePath)
+	}
+
+	private async processData(path: string): Promise<T[]> {
+		const data = await this.read(path, { type: FileType.JSON })
+		if (Array.isArray(data)) return data
+		else return [data]
+	}
+
 	public async load(): Promise<void> {
-		const data = readFilePromise(this.filePath, 'utf8')
-		data.catch(err => {
-			console.error(err)
-		})
-
-		const jsonData: T[] = JSON.parse(await data)
-		if (Array.isArray(jsonData)) {
-			this.lines = jsonData
+		if (this.filePaths.length) {
+			const allLines: any[] = []
+			for (const filePath of this.filePaths) {
+				const data = await this.processData(filePath)
+				allLines.push(...data)
+			}
+			this.lines = allLines
 		} else {
-			this.lines = [jsonData]
+			this.lines = await this.processData(this.filePath)
 		}
 
 		if (this.lines.length === 0) {
 			throw new Error(`JSON file '${this.requestedFilename}' loaded but contains no rows of data.`)
 		}
 
+		if (!this.validStructure(this.lines)) {
+			throw Error(`Data files that have different data structures cannot have the same alias`)
+		}
+
 		this.isLoaded = true
 	}
 
+	public asName(name = ''): void {
+		this.loaderName = name
+	}
+
 	public toString(): string {
 		return `json data ${this.requestedFilename}`
 	}
+
+	public type(): FileType {
+		return FileType.JSON
+	}
 }
 
 export class CSVLoader<T> extends Loader<T> {
-	constructor(public filePath: string, private separator: string = ',', requestedFilename: string) {
-		super(filePath, requestedFilename)
+	constructor(public filePath: string, private separator: string = ',') {
+		super(filePath)
 	}
 
 	public async load(): Promise<void> {
-		let data: string
-		try {
-			data = await readFilePromise(this.filePath, 'utf8')
-		} catch (e) {
-			throw new Error(`unable to read CSV file ${this.filePath}:\ncause: ${e}`)
+		const option = {
+			type: FileType.CSV,
+			delimiter: this.separator,
+			columns: true,
+		}
+		if (this.filePaths.length) {
+			const allLines: any[] = []
```

Comment:
Same as my comment above, please just change nothing and resolve this if I'm wrong
```suggestion
			const allLines: T[] = []
```

Reply:
Resolved